docs/source/models/segment_anything.rst

```diff
@@ -0,0 +1,309 @@
+use it. This boilerplate, is already handle on the high-level API :code:`ImagePrompter`.
+
+Prompter
+^^^^^^^^
+.. _anchor Prompter:
+
+The High level API :code:`ImagePrompter` handle with the image and prompt transformation, preprocessing and prediction for
+a given SAM model.
+
+About the :code:`ImagePrompter`:
+
+#. From a `ModelConfig` loads the desired model with the desired checkpoint to be used as the model to receive the query
+   prompts. For know we just support Segment Anything model, where the *SAM-h* is the default option.
+
+#. Based on the model, the :code:`ImagePrompter` will handle with the necessary transformations to be done into the image
+   and prompts before apply it to the model. These transformations are done using PyTorch backed, by our API of
+   augmentations. Where we use the :class:`kornia.geometry.augmentation.AugmentationSequential` to handle with the different
+   data formats (keypoints, boxes, masks, image).
+
+#. When you use :code:`prompter.set_image(...)`, the prompter will preprocess this image, and then pass it to the encoder,
+   and cache the embeddings to query it after.
+
+    * The preprocess steps are: 1) Resize the image to have its longer side the same size as :code:`image_encoder` image size
+      input. 2) Cache the information of this transformation to apply into the prompts. 3) normalize the image based on the
+      passed mean and standard deviation, or with the values of the SAM dataset. 4) pad on the bottom and right for the image
+      have the encoder expected resolution: :math:`(\text{image_encoder.img_size}, \text{image_encoder.img_size})`.
+
+    * The best image to be used will always have the shape equals to
+      :math:`(\text{image_encoder.img_size}, \text{image_encoder.img_size})`.
+
+#. When you use :code:`prompter.predict(...)`, the prompter will apply the cached transformations on the coordinates of the
+   prompts, and then query this prompts into the cached embeddings.
+
+    * If :code:`output_original_size=True`, the results structure will upsample the logits from it's resolution into the
+      image input original resolution. The output logits has the height and width equals to 256.
+
+#. You can benefit from using the :code:`torch.compile(...)` API (dynamo) for torch >= 2.0.0 version. To compile with dynamo
+   we provide the method :code:`prompter.compile(...)` which will optimize the right parts of the backend model and the
+   prompter itself.
+
+--------------
+
+Example of using the :code:`ImagePrompter`:
+
+Exploring how to simple initialize the :code:`ImagePrompter`, automatically load the weights from a URL,
+read the image and set it to be query, how to write the prompts, and the multiple ways we can use these prompts
+to query the image masks from the SAM model.
+
+
+.. code-block:: python
+
+    import torch
+
+    from kornia.contrib.sam import SamConfig
```

Comment:
actually seeing how is this used and we explain probably it makes sense to move sam under korna.contrib.models ?